Quadriga/src/main/webapp/WEB-INF/views/settings/addpublicpage.jsp

```diff
@@ -6,24 +6,26 @@
 	href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
 <script>
 //@ sourceURL=filename.js
+	
+	
 	$(function() {
-		$("#submit1").click(function(){performAction(this)});
-		$("#submit2").click(function(){performAction(this)});
-		$("#submit3").click(function(){performAction(this)});
+		$("#submit1").click(function(){performAction(this,1)});
+		$("#submit2").click(function(){performAction(this,2)});
+		$("#submit3").click(function(){performAction(this,3)});
 	});
-    
+	
 	function setErrorMsg(obj, field, msg) {
 		$(obj).closest('div.publicpageform').find('#' + field + '_err').html(msg);
 	}
 
-	function performAction(obj) {
+	function performAction(obj,value) {
 		var data = {};
 		var publicpageid = $(obj).closest('div.publicpageform').find('.publicpageid').val();
-		var title = $(obj).closest('div.publicpageform').find('.titleValue').val();
-		var desc = $(obj).closest('div.publicpageform').find('.descValue').val();
-		var order = $(obj).closest('div.publicpageform').find('.orderValue').val();
-		var linkTo = $(obj).closest('div.publicpageform').find('.linkTo').val();
-		var linkText = $(obj).closest('div.publicpageform').find('.linkText').val();
+		var title = $(obj).closest('div.publicpageform').find('#title' + value).val();
+		var desc = $(obj).closest('div.publicpageform').find('#description' + value).val();
+		var order = $(obj).closest('div.publicpageform').find('#orderValue' + value).val();
+		var linkTo = $(obj).closest('div.publicpageform').find('#linkTo' + value).val();
+		var linkText = $(obj).closest('div.publicpageform').find('#linkText' + value).val();
```

Comment:
why do you add that value here? it should have worked before, since you look for certain class of child elements.